### ISSUE TITLE
Vaadin 24 build fails on Alpine

### ISSUE THREAD

**Issue by javier-godoy** [vaadin/flow#15339]
### Description of the bug

I started a Vaadin build *24.0.0.alpha5*, in a system based on Alpine (`jenkins/inbound-agent:3077.vd69cf116da_6f-3-alpine-jdk17`)

The build freezes forever after
```
[INFO] Copying node binary from /root/.vaadin/node/tmp/node-v18.12.1-linux-x64-musl/bin/node to /root/.vaadin/node/bin/node
[INFO] Extracting npm
[INFO] Local node installation successful.
```

It was only after I halted the process, that I was able to see
```
[WARNING] Error checking if node is new enough
com.vaadin.flow.server.frontend.FrontendUtils$UnknownVersionException: Unable to detect version of node. Using command /root/.vaadin/node/node --version
    at com.vaadin.flow.server.frontend.FrontendUtils.getVersion (FrontendUtils.java:1051)
    at com.vaadin.flow.server.frontend.FrontendTools.getNodeVersionAndExecutable (FrontendTools.java:675)
    at com.vaadin.flow.server.frontend.FrontendTools.validateNodeAndNpmVersion (FrontendTools.java:631)
    at com.vaadin.flow.plugin.base.BuildFrontendUtil.prepareFrontend (BuildFrontendUtil.java:143)
    at com.vaadin.flow.plugin.maven.PrepareFrontendMojo.execute (PrepareFrontendMojo.java:56)
```

Moreover, it was not able to detect the version of node because 
```
bash-5.1# /root/.vaadin/node/node --version
Error loading shared library libstdc++.so.6: No such file or directory (needed by /root/.vaadin/node/node)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /root/.vaadin/node/node)
Error relocating /root/.vaadin/node/node: _ZNSt7__cxx1119basic_ostringstreamIcSt11char_traitsIcESaIcEEC1Ev: symbol not found
Error relocating /root/.vaadin/node/node: _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE9push_backEc: symbol not found
Error relocating /root/.vaadin/node/node: _ZStrsIcSt11char_traitsIcESaIcEERSt13basic_istreamIT_T0_ES7_RNSt7__cxx1112basic_stringIS4_S5_T1_EE: symbol not found
Error relocating /root/.vaadin/node/node: _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE6resizeEmc: symbol not found
Error relocating /root/.vaadin/node/node: _ZSt18_Rb_tree_incrementPKSt18_Rb_tree_node_base: symbol not found
```

The same build works fine with `jenkins/inbound-agent:latest-jdk17`

### Expected behavior

The build should fail since `/root/.vaadin/node/node --version` could not be run.

### Minimal reproducible example

Run `mvn package -Pproduction` with a Vaadin 24.0.0.alpha5 application in a system that does not have `libstdc++` installed.

### Versions

- Vaadin / Flow version: 24.0.0.alpha5
- Java version: open jdk 17.0.5+8
- OS version: Linux Alpine 3.16.3

**Comment by javier-godoy** [comment]
**TL;DR `apk add libstdc++ eudev-dev`**

After installing libstdc++, it no longer freezes. However, I got a new error about missing libudev in `build-frontend` goal:

```
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:24.0.0.alpha5:build-frontend (default) on project: Could not execute build-frontend goal: Error occured during goal execution: Unable to load library 'udev':
[ERROR] Error loading shared library libudev.so: No such file or directory
```

**Comment by mcollovati** [comment]
The problem here seems to be that `waitFor` hangs because the STDERR is not read.
The issue seems to happen only on alpine, but perhaps when executing the command we can set `redirectErrorStream(true)` and then read the process input stream before `waitFor`

**Comment by mcollovati** [comment]
`redirectErrorStream(true)` seems not a correct solution for Flow, since we want separate output for STDOUT and STDERR.
We should perhaps asynchronously read the streams separately, and then get the contents after `waitFor`.
This way we consume the streams and `waitFor` will not hang.

**Comment by javier-godoy** [comment]
Fix confirmed with 24.1.0.alpha5. I ran the build in a docker image `FROM jenkins/inbound-agent:latest-alpine-jdk17` (without installing `libstdc++`)

Now the build failure is reported as expected:

```
[ERROR] Command '[/root/.vaadin/node/node, -v]' failed with exit code '127'
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  44.394 s
[INFO] Finished at: 2023-05-11T16:01:14Z
[INFO] ------------------------------------------------------------------------
Waiting for Jenkins to finish collecting data
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:24.1.0.alpha5:prepare-frontend (default) on project addons-demos-24: Could not execute prepare-frontend goal.
[ERROR] 
[ERROR] ======================================================================================================
[ERROR] Vaadin requires Node.js and npm to be installed. The /root/.vaadin directory already contains 'node' but it's either not a file or not a 'node' executable. Please check the /root/.vaadin directory and clean it up: remove '/root/.vaadin/node/node'.
[ERROR] then run the application or Maven goal again.
[ERROR] ======================================================================================================
```

**Comment by mcollovati** [comment]
Thank you for testing. I'll close the issue as it seems the issue if fixed
